src/wallet/rpcdump.cpp

```diff
@@ -479,7 +493,14 @@ UniValue importpubkey(const JSONRPCRequest& request)
     }
     if (fRescan)
     {
-        pwallet->RescanFromTime(TIMESTAMP_MIN, reserver, true /* update */);
+        int64_t scanned_time = pwallet->RescanFromTime(TIMESTAMP_MIN, reserver, true /* update */);
+        if (scanned_time > TIMESTAMP_MIN) {
+            if (pwallet->IsAbortingRescan()) {
```

Comment:
Why is this check within the scope of the `(scanned_time > TIMESTAMP_MIN)` check? Wouldn't the following make more sense:

```cpp
if (pwallet->IsAbortingRescan()) throw JSONRPCError(RPC_MISC_ERROR, "Rescan aborted by user.");
if (scanned_time > TIMESTAMP_MIN) throw JSONRPCError(RPC_WALLET_ERROR, "Rescan was unable to fully rescan the blockchain. Some transactions may be missing.");

Reply:
Yes, done.